### ISSUE TITLE
Add physical forcing and dissipation module (wind stress, bottom drag, Laplacian/biharmonic diffusion)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#75]
## Description

`finitevolX` has **no physical forcing or dissipation utilities**. All three reference repos include these as core components of their RHS computation. Ocean models require:

1. **Wind stress forcing** — surface wind stress curl applied to the top layer
2. **Linear bottom drag** — Ekman-layer friction on the bottom layer
3. **Laplacian diffusion** (∇²) — 2nd-order viscosity for numerical stability
4. **Biharmonic (hyper)diffusion** (∇⁴) — 4th-order diffusion for scale-selective dissipation

### Reference: `louity/qgm_pytorch` — QG forcing/dissipation

**[`QGM.py#L232-L244`](https://github.com/louity/qgm_pytorch/blob/02a6c33839bd720376abf221bc8f33a5612ac190/QGM.py#L232-L244):**
```python
def advection_rhs(self):
    """Advection diffusion RHS for vorticity, only inside domain"""
    rhs = self.jac_coef * self.jacobi_h(self.q_over_f0, self.p)

    # Laplacian diffusion
    p_diff = self.p if self.p_prime is None else self.p - self.p_prime
    delta2_p = self.laplacian_h(p_diff, self.zfbc)
    if self.a_2 != 0.:
        rhs += self.diff_coef * self.laplacian_h_nobc(delta2_p)
    # Biharmonic (hyper)diffusion
    if self.a_4 != 0.:
        rhs -= self.hyperdiff_coef * self.laplacian_h_nobc(
            self.laplacian_h(delta2_p, self.zfbc))

    # Wind forcing (top layer only)
    rhs[...,0:1,:,:] += self.wind_forcing
    # Bottom drag (bottom layer only)
    rhs[...,-1:,:,:] += self.bottom_friction_coef * self.laplacian_h_nobc(
        self.p[...,-1:,:,:])
    return rhs
```

Wind forcing is computed from wind stress curl:
**[`QGM.py#L85-L94`](https://github.com/louity/qgm_pytorch/blob/02a6c33839bd720376abf221bc8f33a5612ac190/QGM.py#L85-L94):**
```python
def curl_wind(tau, dx, dy):
    tau_x, tau_y = tau[...,0], tau[...,1]
    curl_stagg = (tau_y[1:] - tau_y[:-1]) / dx - (tau_x[:,1:] - tau_x[:,:-1]) / dy
    return 0.25*(curl_stagg[:-1,:-1] + curl_stagg[:-1,1:] +
                 curl_stagg[1:,:-1] + curl_stagg[1:,1:])
```

Coefficients:
```python
self.diff_coef = self.a_2 / self.f0**2 / self.dx**4
self.hyperdiff_coef = (self.a_4 / self.f0**2) / self.dx**6
self.bottom_friction_coef = self.delta_ek / (2*abs(self.f0)*self.dx**2*(-self.heights[-1]))
```

### Reference: `louity/qgsw-pytorch` — SWM forcing/dissipation

**[`src/sw.py#L409-L429`](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/sw.py#L409-L429):**
```python
def add_wind_forcing(self, du, dv):
    H_ugrid = (self.h_tot_ugrid) / self.area
    H_vgrid = (self.h_tot_vgrid) / self.area
    du[..., 0,:,:] += self.taux / H_ugrid[...,0,1:-1,:] * self.dx
    dv[..., 0,:,:] += self.tauy / H_vgrid[...,0,:,1:-1] * self.dy
    return du, dv

def add_bottom_drag(self, du, dv):
    du[...,-1,:,:] += -self.bottom_drag_coef * self.u[...,-1,1:-1,:]
    dv[...,-1,:,:] += -self.bottom_drag_coef * self.v[...,-1,:,1:-1]
    return du, dv
```

### Reference: `louity/MQGeometry` — QG FV forcing/dissipation

**[`qgm.py#L186-L202`](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/qgm.py#L186-L202):**
```python
def advection_rhs(self):
    u, v = self.grad_perp(self.psi, self.dx, self.dy)
    div_flux = self.div_flux(self.q, u[...,1:-1,:], v[...,1:-1],
                             self.dx, self.dy, *self.div_flux_args)
    # wind forcing + bottom drag
    omega = self.interp_TP(self.laplacian_h(self.psi, self.dx, self.dy)*self.masks.psi)
    bottom_drag = -self.bottom_drag_coef * omega[...,[-1],:,:]
    if self.nl == 1:
        fcg_drag = self.wind_forcing + bottom_drag
    elif self.nl == 2:
        fcg_drag = torch.cat([self.wind_forcing, bottom_drag], dim=-3)
    else:
        fcg_drag = torch.cat([self.wind_forcing, self.zeros_inside, bottom_drag], dim=-3)
    return (-div_flux + fcg_drag) * self.masks.q
```

## Proposed API

```python
# --- Wind forcing ---
def wind_stress_curl(
    taux: Array, tauy: Array, dx: float, dy: float
) -> Array:
    """Compute wind stress curl ∂τy/∂x − ∂τx/∂y on the T-grid."""

def apply_wind_forcing(
    rhs: Array, curl_tau: Array, H_top: float, layer_axis: int = -3
) -> Array:
    """Add curl(τ)/H₁ to the top layer of the RHS."""

# --- Bottom drag ---
def linear_bottom_drag(
    field: Array, drag_coef: float, layer_axis: int = -3
) -> Array:
    """Compute −r·field for the bottom layer only."""

# --- Diffusion ---
def laplacian_diffusion(
    field: Array, coef: float, dx: float, dy: float
) -> Array:
    """Compute ν₂·∇²(field)."""

def biharmonic_diffusion(
    field: Array, coef: float, dx: float, dy: float
) -> Array:
    """Compute −ν₄·∇⁴(field) (scale-selective hyperdiffusion)."""
```

## Acceptance Criteria

- [ ] `wind_stress_curl(taux, tauy, dx, dy)` function
- [ ] `apply_wind_forcing(rhs, curl_tau, H_top)` targeting only top layer
- [ ] `linear_bottom_drag(field, drag_coef)` targeting only bottom layer
- [ ] `laplacian_diffusion(field, coef, dx, dy)` — 2nd-order viscosity
- [ ] `biharmonic_diffusion(field, coef, dx, dy)` — 4th-order hyperdiffusion
- [ ] All functions mask-aware (multiply by domain mask)
- [ ] Multi-layer support (wind on layer 0, drag on layer -1, diffusion on all)
- [ ] Tests verifying forcing/dissipation magnitudes against analytic cases
- [ ] Tests verifying energy injection (wind) and dissipation (drag, diffusion) signs
- [ ] JAX-compatible, `jit`-able

## Priority
**High** — no ocean model can run without forcing and dissipation.

---

_References:_
- _[louity/qgm_pytorch/QGM.py — advection_rhs](https://github.com/louity/qgm_pytorch/blob/02a6c33839bd720376abf221bc8f33a5612ac190/QGM.py#L232-L244)_
- _[louity/qgsw-pytorch/src/sw.py — add_wind_forcing, add_bottom_drag](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/sw.py#L409-L429)_
- _[louity/MQGeometry/qgm.py — advection_rhs](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/qgm.py#L186-L202)_
